ark/type/parser/objectLiteral.ts

```diff
@@ -46,32 +95,46 @@ export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
 			})
 		}
 		ctx.path.pop()
+
+		hasSeenFirstKey ||= true
 	}
+
 	return schema({
 		basis: "object",
 		required,
 		optional
 	})
 }
 
+/**
+ * Infers the contents of an object literal, ignoring a spread definition
+ * You probably want to use {@link inferObjectLiteral} instead.
+ */
+type inferObjectLiteralInner<def extends object, $, args> = {
+	// since def is a const parameter, we remove the readonly modifier here
+	// support for builtin readonly tracked here:
+	// https://github.com/arktypeio/arktype/issues/808
+	-readonly [k in keyof def as nonOptionalKeyFrom<
```

Comment:
The `...` is ignored because `nonOptionalKeyFrom` and `optionalKeyFrom` only read those nodes with the `required` or `optional` kinds, and hence not the `spread` kind.